content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/Methods.class/__methods__/amazon_customizerequest.rb

```diff
@@ -2,99 +2,166 @@
+                  customization_template = nil
+
+                  unless customization_template_search_by_function.nil?
+                    # Search for customization templates enabled for Cloud-Init that equal MiqProvisionAmazon_prov.get_tags[:function]
+                    if customization_template.blank?
+                      log(:info, "Searching for customization templates (Cloud-Init) enabled that are named: %{function}" % {
+                        :function => customization_template_search_by_function
+                      })
+                      customization_template = prov.eligible_customization_templates.detect do |ct|
+                        ct.name.casecmp(customization_template_search_by_function) == 0
+                      end
+                    end
+                  end
+                  unless customization_template_search_by_template_name.nil?
+                    # Search for customization templates enabled for Cloud-Init that match the template/image name
+                    if customization_template.blank?
+                      log(:info, "Searching for customization templates (Cloud-Init) enabled that are named: %{name}" % {
+                        :name => customization_template_search_by_template_name
+                      })
+                      customization_template = prov.eligible_customization_templates.detect do |ct|
+                        ct.name.casecmp(customization_template_search_by_template_name) == 0
+                      end
+                    end
+                  end
+                  unless customization_template_search_by_ws_values.nil?
+                    # Search for customization templates enabled for Cloud-Init that match ws_values[:customization_template]
+                    if customization_template.blank?
+                      log(:info, "Searching for customization templates (Cloud-Init) enabled that are named: %{ws_value}" % {
+                        :ws_value => customization_template_search_by_ws_values
+                      })
+                      customization_template = prov.eligible_customization_templates.detect do |ct|
+                        ct.name.casecmp(customization_template_search_by_ws_values) == 0
+                      end
+                    end
+                  end
+                  if customization_template.blank?
+                    log(:warn, "Failed to find matching Customization Template", true)
+                  else
+                    log(:info, "Found Customization Template ID: %{id} Name: %{name} Description: %{desc}" % {
+                      :id   => customization_template.id,
+                      :name => customization_template.name,
+                      :desc => customization_template.description
+                    })
+                    prov.set_customization_template(customization_template) rescue nil
```

Comment:
This is the only one I'm not really sure about and may want to keep in this refactoring and address later.